lib/src/builders/slash_command_builder.dart

```diff
@@ -2,26 +2,70 @@ import 'package:nyxx/nyxx.dart';
 
 import 'package:nyxx_interactions/src/builders/command_option_builder.dart';
 import 'package:nyxx_interactions/src/builders/command_permission_builder.dart';
-
 import 'package:nyxx_interactions/src/models/slash_command_type.dart';
 import 'package:nyxx_interactions/src/models/command_option.dart';
 import 'package:nyxx_interactions/src/interactions.dart';
 import 'package:nyxx_interactions/src/internal/utils.dart';
 import 'package:nyxx_interactions/src/typedefs.dart';
 
-/// A slash command, can only be instantiated through a method on [Interactions]
+/// A slash command, can only be instantiated through a method on [IInteractions]
 class SlashCommandBuilder extends Builder {
   /// The commands ID that is defined on registration and used for permission syncing.
   late final Snowflake _id;
 
   /// Command name to be shown to the user in the Slash Command UI
   final String name;
 
+  /// The command names to be shown to the user in the Slash Command UI by specified locales.
+  /// See the [available locales](https://discord.com/developers/docs/reference#locales) for a list of available locales.
+  /// The key is the locale and the value is the name of the command in that locale.
+  /// Values follow the same constraints as [name] (`^[\w-]{1,32}$`).
+  ///
+  /// An example:
+  /// {@template slashcommand.builder.example}
+  /// ```dart
+  /// final scb = SlashCommandBuilder(
+  ///   'hello',
+  ///   'Hello World!',
+  ///   [],
+  ///   localizationsName: {
+  ///     'fr': 'salut',
+  ///     'de': 'hallo',
+  ///   },
+  ///   localizationsDescription: {
+  ///     'fr': 'Salut le monde !',
+  ///     'de': 'Hallo Welt!',
+  ///   },
+  /// );
+  /// ```
+  /// {@endtemplate}
+  /// {@template discord.override.link}
+  ///
+  /// At this time, the localized feature is still in developement, so you you must set [an override] on your discord client.
+  ///
+  /// The override is `feature/web-slash-commands-localization` that you can get [here].
+  ///
+  /// [an override]: https://support.discord.com/hc/de/articles/360030114991
+  /// [here]: https://discord.com/__development/link?s=qTpnDQLUm66XsoBawqlEcXO3G9jPncSIJy9OSKnIcSk%3D.eyJ0YXJnZXRCdWlsZE92ZXJyaWRlIjp7ImRpc2NvcmRfd2ViIjp7InR5cGUiOiJicmFuY2giLCJpZCI6ImZlYXR1cmUvd2ViLXNsYXNoLWNvbW1hbmQtbG9jYWxpemF0aW9uIn19LCJyZWxlYXNlQ2hhbm5lbCI6bnVsbCwidmFsaWRGb3JVc2VySWRzIjpbXSwiYWxsb3dMb2dnZWRPdXQiOmZhbHNlLCJleHBpcmVzQXQiOiJGcmksIDIwIE1heSAyMDIyIDE4OjA2OjU5IEdNVCJ9
+  /// {@endtemplate}
+  ///
+  final Map<String, String>? localizationsName;
```

Comment:
The locale should be specified as an enum (`Map<Locale, String>` instead of `Map<String, String>`). Same for  all other localisation options.

Feel free to bump the SDK version to 2.17 and use the new enhanced enums.